gm4_heart_canisters/data/heart_canisters/functions/update.mcfunction

```diff
@@ -0,0 +1,14 @@
+scoreboard players operation @s heart_can_time -= @s heart_can_health
+scoreboard players set @s[scores={heart_can_time=0}] heart_can_time 1337
+effect clear @s[scores={heart_can_time=..1336}] health_boost
+effect give @s[scores={heart_can_time=..1336,heart_can_health=1}] health_boost 1000000 0 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=2}] health_boost 1000000 1 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=3}] health_boost 1000000 2 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=4}] health_boost 1000000 3 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=5}] health_boost 1000000 4 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=6}] health_boost 1000000 5 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=7}] health_boost 1000000 6 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=8}] health_boost 1000000 7 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=9}] health_boost 1000000 8 true
+effect give @s[scores={heart_can_time=..1336,heart_can_health=10}] health_boost 1000000 9 true
+scoreboard players operation @s heart_can_time = @s heart_can_health
```

Comment:
So I'm partially confused with what you're doing here. I've figured out that it only tries to apply the health boost if your number of heart canisters has changed, but why try and figure that out in the first place? It looks to me that since its not constantly applying the effect, when the duration expires, the effect won't be reapplied to the player until they move their heart-canisters again. And why 1336? It dosen't seem to represent anything. The highest heart_can_time should ever be is 10.

Reply:
I honestly don't know it either but that's how it was in the 1.12 version